src/Blocks/BlockButton.h

```diff
@@ -113,15 +122,36 @@ class cBlockButtonHandler :
 		}
 	}
 
+	static bool CanBePlacedOn(BLOCKTYPE a_BlockType, NIBBLETYPE a_BlockMeta, eBlockFace a_BlockFace)
+	{
+		switch(a_BlockType)
+		{
+			case E_BLOCK_HOPPER:
```

Comment:
For example, vanilla behavior allows buttons to be placed only on top of hoppers, not on other sides